node/p2p/misc.go

```diff
@@ -0,0 +1,64 @@
+package p2p
+
+import (
+	"time"
+
+	"github.com/ipfs/go-datastore"
+	connmgr "github.com/libp2p/go-libp2p-connmgr"
+	coreconnmgr "github.com/libp2p/go-libp2p-core/connmgr"
+	"github.com/libp2p/go-libp2p-core/peerstore"
+	"github.com/libp2p/go-libp2p-peerstore/pstoremem"
+	"github.com/libp2p/go-libp2p/p2p/net/conngater"
+)
+
+// ConnManagerConfig configures connection manager.
+type ConnManagerConfig struct {
+	// Low and High are watermarks governing the number of connections that'll be maintained.
+	Low, High int
+	// GracePeriod is the amount of time a newly opened connection is given before it becomes subject to pruning.
+	GracePeriod time.Duration
+}
+
+// DefaultConnManagerConfig returns defaults for ConnManagerConfig.
+func DefaultConnManagerConfig() *ConnManagerConfig {
+	return &ConnManagerConfig{
+		Low:         50,
+		High:        100,
+		GracePeriod: time.Minute,
+	}
+}
+
+// ConnectionManager provides a constructor for ConnectionManager.
+func ConnectionManager(cfg *Config) func() (coreconnmgr.ConnManager, error) {
+	return func() (coreconnmgr.ConnManager, error) {
+		fpeers, err := cfg.mutualPeers()
+		if err != nil {
+			return nil, err
+		}
+
+		bpeers, err := cfg.bootstrapPeers()
+		if err != nil {
+			return nil, err
+		}
+
+		cm := connmgr.NewConnManager(cfg.ConnManager.Low, cfg.ConnManager.High, cfg.ConnManager.GracePeriod)
+		for _, info := range fpeers {
+			cm.Protect(info.ID, "69protected69_config")
```

Comment:
what is this hardcoded string?

Reply:
It is a tag, like a comment of why this connection/peer is protected. It won't be reused in this case so I kept it anonymous.

Reply:
Is the tag necessary? If so, maybe we should make a comment about what this `69protected69_X` thing means lol

Reply:
There can be arbitrary strings.

Reply:
maybe we should have something more meaningful then? Or just pass empty? What does `69protected69` mean lol

Reply:
Updated to be more meaningfull

Reply:
Btw, always keep it 69 and unprotected 
![image](https://user-images.githubusercontent.com/22449364/133616904-c2f0cb9f-390c-49fe-9615-b0d626b5e08e.png)

Reply:
Shouldn't the tag reflect why this is protected? Something like "mutual_peer" makes more sense than `69protected69_config`. Or is it common to actually use random strings?

Reply:
Lol, you were faster. Thanks for resolving

Reply:
That's a very tiny and unimportant thing to spend our energy to, so let's stop here and I updated this to be more meaningful.